operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java

```diff
@@ -173,7 +179,8 @@ public int getAttemptCount() {
 
           @Override
           public boolean isLastAttempt() {
-            return controller.getConfiguration().getRetry() == null;
+            return controller.getConfiguration().getRetry() == null ||
+                retryConfigurationHasZeroAttempts; // check also if the retry is limited to 0
```

Comment:
I would just reverse the order of the check for optimisation purposes… (not a big deal, obviously 😅)

Reply:
makes sense, done!